server/server.js

```diff
@@ -138,4 +137,92 @@ app.get('/sidebar/all', (req, res) => {
   });
 });
 
+// Create
+app.post('/sidebar/all', async (req, res) => {
+  console.log('POST request to /sidebar/all');
+  const newDocument = req.body;
+  const newCourseId = newDocument.courseId;
+  console.log(newDocument);
+  // change string date to date type
+  newDocument.price.saleEndDate = new Date(newDocument.price.saleEndDate);
+
+  if (typeof newCourseId !== 'number') {
+    res.status(400).send('Sorry, invalid request: courseId is not a number');
```

Comment:
Personally I find /sidebar/all somewhat confusing. Assuming you'd like to create another record and all the services for Udemy are consistent on the way passing parameter in url, I was expecting to see app.post('/Create/sidebar/item/?courseId=x', ....).

Reply:
I thought that using a /create at the beginning of the endpoint would be redundant since a POST request is being made. It would be inferred through the type of request what the endpoint would do. I think we can discuss further as a team to see which direction we want to move in